pageserver/src/layered_repository.rs

```diff
@@ -111,7 +111,7 @@ pub struct LayeredRepository {
     timelines: Mutex<HashMap<ZTimelineId, Arc<LayeredTimeline>>>,
 
     walredo_mgr: Arc<dyn WalRedoManager + Send + Sync>,
-    relish_uploader: Option<Arc<QueueBasedRelishUploader>>,
+    storage_uploader: Option<Arc<StorageUploader>>,
```

Comment:
It used to be the previous naive way to start relish storage sync place: all `QueueBasedRelishUploader` spun a thread inside with their own queue.

This is utterly wrong IMO and now the `StorageUploader` masks a method for inserting uploads into the global queue that's polled by the main upload/download thread loop.

The uploader is stored in every repo (`tenant_mgr` does that) and then propagated into every layer where is eventually used to upload timelines: 
https://github.com/zenithdb/zenith/pull/686/files?authenticity_token=Gvx%2FdzDAaLQeUmjPTZIHdory2BX2C4%2BQg6w0ronyYfNT5Lykzja61ISAvx6MKf7RrGsc0qvvb9hHG075Ilir%2FA%3D%3D&file-filters%5B%5D=.rs&hide-deleted-files=true#diff-2dee9987054422607c1ab7369e102927d8477dea6d75c4c7f3891e3d65b15d1eR259